### PR TITLE
Add LocalFileManagerDriver

### PR DESCRIPTION
**Pull request by dylanholmes**
Changes:

- Add `BaseFileManagerDriver`
- Add `LocalFileManagerDriver`
- Remove all existing fields from `FileManager`.
- Add `FileManager.file_manager_driver`
- Added optional `BaseLoader.encoding` field. This allows the `LocalFileManagerDriver` to save files with the same encoding as is used for loading them. This is possible without modifying `BaseLoader`, but not doing so would be more fragile and awkward.
- Add `BlobLoader`

### CODE REVIEW

**Comment by dylanholmes** [comment]
Reverted to draft to make some changes I realized I needed+wanted to make while testing:
- `LocalFileManagerDriver. load_file` needs to read in the file after the changes rebased that removed filepaths from the source of the loaders.
- I want to move the usage of the loaders into the `BaseFileManager`, then have the subclasses implement a method for that reads/writes bytes only.

File: griptape/drivers/file_manager/base_file_manager_driver.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+class BaseFileManagerDriver(ABC):
+    """
+    BaseFileManagerDriver can be used to list, load, and save files.
+
+    Attributes:
+        loaders: Dictionary of file extensions and matching loaders to use when loading files in load_files_from_disk.
+        default_loader: The loader to use when loading files in load_files_from_disk without any matching loader in `loaders`.
+    """
+
+    default_loader: Optional[BaseLoader] = field(default=None, kw_only=True)
```
[the remark this answers is not in the document]
Comment (dylanholmes) [reply griptape/drivers/file_manager/base_file_manager_driver.py:19]: Let's say you want know you will only be loading plain text files, but they may or may not have the extension `.txt` (I do this all the time). Then you'd set `loaders` to `{}` and `default_loader` to `TextLoader()`.

[the remark this answers is not in the document]
Comment (dylanholmes) [reply griptape/drivers/file_manager/base_file_manager_driver.py:19]: How about creating a `BlobLoader` that just takes in `bytes` as a source and returns a `BlobArtifact`, and keeping the name `default_loader`?

File: griptape/drivers/file_manager/base_file_manager_driver.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Optional
+from attr import Factory, define, field
+from griptape.artifacts import BaseArtifact, ErrorArtifact, TextArtifact, InfoArtifact, ListArtifact
+import griptape.loaders as loaders
```
Comment (dylanholmes) [inline griptape/drivers/file_manager/base_file_manager_driver.py:5]: This import alias is used to resolve a circular dependency between the drivers and loaders illustrated by the error shown below. Would be nice to have a more elegant solution, but I haven't found one yet.

```
_________________________________________________________ ERROR collecting tests/unit/loaders/test_blob_loader.py __________________________________________________________
ImportError while importing test module '/Users/dylan/Documents/griptape/tests/unit/loaders/test_blob_loader.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/unit/loaders/test_blob_loader.py:2: in <module>
    from griptape.loaders.blob_loader import BlobLoader
griptape/loaders/__init__.py:2: in <module>
    from .base_text_loader import BaseTextLoader
griptape/loaders/base_text_loader.py:11: in <module>
    from griptape.drivers import BaseEmbeddingDriver
griptape/drivers/__init__.py:96: in <module>
    from .file_manager.base_file_manager_driver import BaseFileManagerDriver
griptape/drivers/file_manager/base_file_manager_driver.py:6: in <module>
    from griptape.loaders import BaseLoader, BlobLoader, CsvLoader, ImageLoader, PdfLoader, TextLoader
E   ImportError: cannot import name 'BlobLoader' from partially initialized module 'griptape.loaders' (most likely due to a circular import) (/Users/dylan/Documents/griptape/griptape/loaders/__init__.py)
```

File: griptape/drivers/file_manager/base_file_manager_driver.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dylanholmes) [reply griptape/drivers/file_manager/base_file_manager_driver.py]: Added `BlobLoader`, set it as the default for `default_loader`, left the name as `default_loader`, and removed `Optional`.

File: griptape/drivers/file_manager/base_file_manager_driver.py (hunk 2 above)
Comment (collindutter) [reply griptape/drivers/file_manager/base_file_manager_driver.py:5]: The `griptape.drivers` module is becoming quite large and causing more and more circular dependency issues as we integrate Drivers in different parts of the framework. I guess if this works it works.

**Pull request merged**